plugins/cpp/parser/src/cppparser.cpp

```diff
@@ -329,19 +332,137 @@ int CppParser::worker(const clang::tooling::CompileCommand& command_)
+            odb::query<model::CppAstNode>::astType == model::CppAstNode::AstType::Definition);
+          for (const model::CppAstNode &astNode : defCppAstNodes)
+          {
+            // Delete CppEntity
+            auto delCppEntities = _ctx.db->query<model::CppEntity>(
+              odb::query<model::CppEntity>::astNodeId == astNode.id);
+            for (const model::CppEntity &entity : delCppEntities)
+            {
+              _ctx.db->erase<model::CppEntity>(entity.id);
+            }
+
+            // Delete CppInheritance
+            auto delCppInheritance = _ctx.db->query<model::CppInheritance>(
+              odb::query<model::CppInheritance>::derived == astNode.mangledNameHash);
+            for (const model::CppInheritance &inheritance : delCppInheritance)
+            {
+              _ctx.db->erase<model::CppInheritance>(inheritance.id);
+            }
+
+            // Delete CppFriendship
+            auto delCppFriendship = _ctx.db->query<model::CppFriendship>(
+              odb::query<model::CppFriendship>::target == astNode.mangledNameHash);
+            for (const model::CppFriendship &friendship : delCppFriendship)
+            {
+              _ctx.db->erase<model::CppFriendship>(friendship.id);
+            }
+
+            // Delete CppNode (connected to CppAstNode) with all its connected CppNodes
+            auto delNodes = _ctx.db->query<model::CppNode>(
+              odb::query<model::CppNode>::domainId == std::to_string(astNode.id) &&
+              odb::query<model::CppNode>::domain == model::CppNode::CPPASTNODE);
+            for (model::CppNode &node : delNodes)
+            {
+              for (model::CppNodeId nodeId : collectNodeSet(node.id))
+              {
+                _ctx.db->erase<model::CppNode>(nodeId);
+              }
+            }
+          }
+
+          // Delete BuildAction
+          auto delSources = _ctx.db->query<model::BuildSource>(
+            odb::query<model::BuildSource>::file == delFile->id);
+          for (const model::BuildSource &source : delSources)
+          {
+            _ctx.db->erase<model::BuildAction>(source.action->id);
+          }
+
+          // Delete CppNode (connected to File) with all its connected CppNodes
+          auto delNodes = _ctx.db->query<model::CppNode>(
+            odb::query<model::CppNode>::domainId == std::to_string(delFile->id) &&
+            odb::query<model::CppNode>::domain == model::CppNode::FILE);
+          for (model::CppNode &node : delNodes)
+          {
+            for (model::CppNodeId nodeId : collectNodeSet(node.id))
+            {
+              _ctx.db->erase<model::CppNode>(nodeId);
+            }
+          }
+
+          break;
+        }
+
+        case IncrementalStatus::ADDED:
+          // Empty deliberately
+          break;
+      }
+    }
+  }); // end of transaction
+}
+
 bool CppParser::parse()
 {
+  if(_ctx.options.count("incremental"))
```

Comment:
This is a leftover from the previous iteration where incrementality needed to be explicitly toggled, right? Mark a TODO here, please.

Reply:
There was no previous iteration, this is the first, therefore contains some changes that were later dropped according to the project meeting discussions. E.g. the `--incremental` flag was first decided to be introduced, but later to be removed as CodeCompass should detect this automatically.
I advise to see the final changeset, not the single commits, the overall amount of changed codelines is not that much.

So, to clarify it: this is a `feature branch` (managed according to the [Git Feature Branching Model](https://nvie.com/posts/a-successful-git-branching-model/)), which is about to get merged into the `develop` branch (`master` in this case, because there is no `develop`).
Therefore it contains all changes of the feature branch, as normal.

Reply:
Yes, must have remained there from the previous iteration, when the `--incremental` flag was introduced, but later it was decided to be dropped and to automatically perform an incremental parsing when required.
The whole condition can be removed, as it just prints out debug information.